pkg/osquery/extension.go

```diff
@@ -418,35 +418,38 @@ func (e *Extension) Enroll(ctx context.Context) (string, bool, error) {
 	// We used to see the enrollment details fail, but now that we're running as an exec,
 	// it seems less likely. Try a couple times, but backoff fast.
 	var enrollDetails service.EnrollmentDetails
-	if osqPath := e.knapsack.LatestOsquerydPath(ctx); osqPath == "" {
-		e.slogger.Log(ctx, slog.LevelInfo,
-			"skipping enrollment details, no osqueryd path, this is probably CI",
-		)
-		span.AddEvent("skipping_enrollment_details")
-	} else {
-		if err := backoff.WaitFor(func() error {
-			enrollDetails, err = getEnrollDetails(ctx, osqPath)
-			if err != nil {
-				e.slogger.Log(ctx, slog.LevelDebug,
-					"getEnrollDetails failed in backoff",
-					"err", err,
-				)
-			}
-			return err
-		}, 30*time.Second, 5*time.Second); err != nil {
-			if os.Getenv("LAUNCHER_DEBUG_ENROLL_DETAILS_REQUIRED") == "true" {
-				return "", true, fmt.Errorf("query enrollment details: %w", err)
-			}
+	osqPath := e.knapsack.LatestOsquerydPath(ctx)
 
-			e.slogger.Log(ctx, slog.LevelError,
-				"failed to get enrollment details with retries, moving on",
+	if err := backoff.WaitFor(func() error {
+		enrollDetails, err = getEnrollDetails(ctx, osqPath)
+		switch {
+		case errors.Is(err, osqueryNotFoundErr{}):
+			e.slogger.Log(ctx, slog.LevelDebug,
+				"osquery not found, using runtime values for enrollment",
 				"err", err,
 			)
-			traces.SetError(span, fmt.Errorf("query enrollment details: %w", err))
-		} else {
-			span.AddEvent("got_enrollment_details")
+			return nil
+		case err != nil:
+			e.slogger.Log(ctx, slog.LevelDebug,
+				"getEnrollDetails failed in backoff",
+				"err", err,
+			)
+		}
+		return err
+	}, 30*time.Second, 5*time.Second); err != nil {
+		if os.Getenv("LAUNCHER_DEBUG_ENROLL_DETAILS_REQUIRED") == "true" {
+			return "", true, fmt.Errorf("query enrollment details: %w", err)
 		}
+
+		e.slogger.Log(ctx, slog.LevelError,
+			"failed to get enrollment details with osquery, using runtime values for enrollment",
```

Comment:
I'm confused by this log -- it makes it sound like we're about to fall back to setting `enrollDetails` another way, but then we don't do that. Are we supposed to be performing another operation here? Or does the log message mean "failed to get enrollment details via osquery/runtime"...?